spec/helpers/registration_helper_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'spec_helper'
+
+describe RegistrationHelper do
+  describe '#state_options_query' do
+    let(:query_with_abbr) do
+      's.id as s.abbr for s in enterprise.country.states'
+    end
+    let(:query_with_name) do
+      's.id as s.name for s in enterprise.country.states'
+    end
+
+    subject { helper.state_options_query }
+
+    context 'when the `STATE_OPTION_FIELD` is not defined' do
+      let!(:original_var_state) { ENV.delete('STATE_OPTION_FIELD') }
+
+      after { ENV['STATE_OPTION_FIELD'] = original_var_state }
```

Comment:
I prefer to use `around` for these cases. It makes it a bit more clear that a `before/let` and a `after` depend on each other.